pkg/controller/v1beta1/inferenceservice/components/explainer.go

```diff
@@ -0,0 +1,98 @@
+
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+)
+
+var _ Component = &Explainer{}
+
+// Explainer reconciles resources for this component.
+type Explainer struct {
+	client                 client.Client
+	scheme                 *runtime.Scheme
+	inferenceServiceConfig *v1beta1.InferenceServicesConfig
+	credentialBuilder      *credentials.CredentialBuilder
+	Log                    logr.Logger
+}
+
+func NewExplainer(client client.Client, scheme *runtime.Scheme, inferenceServiceConfig *v1beta1.InferenceServicesConfig) Component {
+	return &Explainer{
+		client:                 client,
+		scheme:                 scheme,
+		inferenceServiceConfig: inferenceServiceConfig,
+		Log:                    ctrl.Log.WithName("ExplainerReconciler"),
+	}
+}
+
+// Reconcile observes the world and attempts to drive the status towards the desired state.
+func (p *Explainer) Reconcile(isvc *v1beta1.InferenceService) error {
+	p.Log.Info("Reconciling Explainer", "ExplainerSpec", isvc.Spec.Explainer)
+	explainer := isvc.Spec.Explainer.GetImplementation()
+	annotations := utils.Filter(isvc.Annotations, func(key string) bool {
+		return !utils.Includes(constants.ServiceAnnotationDisallowedList, key)
+	})
+	// KNative does not support INIT containers or mounting, so we add annotations that trigger the
+	// StorageInitializer injector to mutate the underlying deployment to provision model data
+	if sourceURI := explainer.GetStorageUri(); sourceURI != nil {
+		annotations[constants.StorageInitializerSourceUriInternalAnnotationKey] = *sourceURI
+	}
+	objectMeta := metav1.ObjectMeta{
+		Name:      isvc.Name + "-" + string(v1beta1.ExplainerComponent),
+		Namespace: isvc.Namespace,
+		Labels: utils.Union(isvc.Labels, map[string]string{
+			constants.InferenceServicePodLabelKey: isvc.Name,
+			constants.KServiceComponentLabel:      string(v1beta1.ExplainerComponent),
+		}),
+		Annotations: annotations,
+	}
+	if len(isvc.Spec.Explainer.PodSpec.Containers) == 0 {
+		container := explainer.GetContainer(isvc.ObjectMeta, isvc.Spec.Explainer.GetExtensions(), p.inferenceServiceConfig)
+		isvc.Spec.Explainer.PodSpec = v1beta1.PodSpec{
+			Containers: []v1.Container{
+				*container,
+			},
+		}
+	} else {
+		container := explainer.GetContainer(isvc.ObjectMeta, isvc.Spec.Explainer.GetExtensions(), p.inferenceServiceConfig)
+		isvc.Spec.Explainer.PodSpec.Containers[0] = *container
+	}
+	// Here we allow switch between knative and vanilla deployment
+	podSpec := v1.PodSpec(isvc.Spec.Explainer.PodSpec)
+	r := knative.NewKsvcReconciler(p.client, p.scheme, objectMeta, &isvc.Spec.Explainer.ComponentExtensionSpec,
+		&podSpec, isvc.Status.Components[v1beta1.ExplainerComponent])
+
+	if err := controllerutil.SetControllerReference(isvc, r.Service, p.scheme); err != nil {
+		return err
+	}
+	if status, err := r.Reconcile(); err != nil {
+		return err
+	} else {
+		isvc.Status.PropagateStatus(v1beta1.ExplainerComponent, status)
```

Comment:
This if else structure will need to be changed if we change this code. I'd follow the more idiomatic
```
if status, err := r.Reconcile(); err != nil {
	return err
}
isvc.Status.PropagateStatus(v1beta1.ExplainerComponent, status)
return nil
```

Reply:
done